models/infrastructure_edge.py

```diff
@@ -1577,24 +1608,47 @@ async def run(client: InfrahubClient, log: logging.Logger, branch: str):
             management_pool=management_pool,
             external_pool=external_pool,
         )
-        log.debug(f"{response} - Creation Completed")
+        log.info(f"{response} - Creation Completed")
 
     # ------------------------------------------
     # Add profile on interfaces upstream/backbone
     # ------------------------------------------
     # FIXME - Could do it better
+    log.info("Starting to apply profiles to interfaces")
     upstream_interfaces = await client.filters(branch=branch, kind="InfraInterfaceL3", role__value="upstream")
     backbone_interfaces = await client.filters(branch=branch, kind="InfraInterfaceL3", role__value="backbone")
+    upstream_profile = store.get(key="upstream_profile", kind="ProfileInfraInterfaceL3")
+    backbone_profile = store.get(key="backbone_profile", kind="ProfileInfraInterfaceL3")
+
+    batch = await client.create_batch()
+    backbone_profiles = []
+    upstream_profiles = []
+
+    # Fetch interface profiles
     for interface in upstream_interfaces:
-        profile_id = store.get(key="upstream_profile", kind="ProfileInfraInterfaceL3").id
-        await interface.profiles.fetch()
-        interface.profiles.add(profile_id)
-        await interface.save()
+        batch.add(task=interface.profiles.fetch, node=interface)
+        upstream_profiles.append(interface)
     for interface in backbone_interfaces:
-        profile_id = store.get(key="backbone_profile", kind="ProfileInfraInterfaceL3").id
-        await interface.profiles.fetch()
-        interface.profiles.add(profile_id)
-        await interface.save()
+        batch.add(task=interface.profiles.fetch, node=interface)
+        backbone_profiles.append(interface)
+
+    async for _, response in batch.execute():
+        log.debug(f"{response} - Creation Completed")
```

Comment:
this one seems like more of a ` - Fetch Completed` or something than a ` - Creation Completed`

Reply:
Ah yes thanks. I'll merge it as is now and fix this in the next iteration as it's only at the debug level anyway. We really need to get rid of these fetch operations all together and include them in the original query..